### ISSUE TITLE
AssertionError in compiling a simple pipeline containing dsl.Condition

### ISSUE THREAD

**Issue by deepio-oc** [kubeflow/pipelines#2295]
I am getting a AssertionError in compiling a simple pipeline with dsl.Condition. Following is the code:

```python
def skeleton_pipeline(counter = 0):

    a_op = comp.func_to_container_op(FileCheck, base_image=base_image)
    a_task = a_op (counter)

    with dsl.Condition(a_task.outputs['status'] == -1):
        b_op = comp.func_to_container_op(NoFileToIngest, base_image=base_image)
        b_task = b_op (1)
````

I get following assertion:

pipelines/sdk/python/kfp/compiler/_data_passing_rewriter.py", line 146, in fix_big_data_passing
    assert argument_placeholder_parts[2] == 'outputs'
AssertionError

**Comment by Ark-kun** [comment]
Thanks a lot for discovering this bug and fixing it!

**Comment by Ark-kun** [comment]
Fixed by #2297
